Tribler goes to foreground when Magnet:: is clicked and download is started.

Now it looks like nothing is happening, other clients also pop to the foreground.

I have just tried the functionality a few times with several different magnet links, turns out that some of them may not be parsed correctly and they don't have infohash, so it raises an exception as follows:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/wx-2.8-gtk2-unicode/wx/_core.py", line 14665, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/tribler_main.py", line 1079, in start_asked_download
    self.frame.startDownloadFromMagnet(torrentfilename)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Main/vwxGUI/MainFrame.py", line 424, in startDownloadFromMagnet
    tdef = TorrentDefNoMetainfo(infohash, name, url=url)
  File "/home/lfei/workspace/tud/p2p/tribler/Tribler/Core/TorrentDef.py", line 1110, in __init__
    assert isinstance(infohash, str), "INFOHASH has invalid type: %s" % type(infohash)
AssertionError: INFOHASH has invalid type: <type 'NoneType'>
```

@whirm I will add a `try`-`except` there and show an error dialog if something goes wrong so that users can know what is happening.

--- a/Tribler/Main/vwxGUI/MainFrame.py
+++ b/Tribler/Main/vwxGUI/MainFrame.py
@@ -418,8 +418,15 @@ class MainFrame(wx.Frame):
         name, infohash, _ = parse_magnetlink(url)
         if name is None:
             name = ""
-        tdef = TorrentDefNoMetainfo(infohash, name, url=url)
-        wx.CallAfter(self.startDownload, tdef=tdef, cmdline=cmdline, destdir=destdir, selectedFiles=selectedFiles, vodmode=vodmode, hops=0)
+        try:
+            tdef = TorrentDefNoMetainfo(infohash, name, url=url)
+            wx.CallAfter(self.startDownload, tdef=tdef, cmdline=cmdline, destdir=destdir, selectedFiles=selectedFiles, vodmode=vodmode, hops=0)
+        except Exception, e:
+            # show an error dialog
+            dlg = wx.MessageBox(self, "Failed to add magnet download: %s" % str(e),
+                                "Failed to add magnet download", wx.OK | wx.ICON_ERROR)
+            dlg.ShowModal()
+            dlg.Destroy()
         return True
 
     def startDownloadFromUrl(self, url, destdir=None, cmdline=False, selectedFiles=None, vodmode=False, hops=0):

--- a/Tribler/Main/vwxGUI/MainFrame.py
+++ b/Tribler/Main/vwxGUI/MainFrame.py
@@ -419,12 +419,14 @@ class MainFrame(wx.Frame):
         if name is None:
             name = ""
         try:
+            if infohash is None:
+                raise RuntimeError("Missing infohash")
             tdef = TorrentDefNoMetainfo(infohash, name, url=url)
             wx.CallAfter(self.startDownload, tdef=tdef, cmdline=cmdline, destdir=destdir, selectedFiles=selectedFiles, vodmode=vodmode, hops=0)
         except Exception, e:
             # show an error dialog
-            dlg = wx.MessageBox(self, "Failed to add magnet download: %s" % str(e),
-                                "Failed to add magnet download", wx.OK | wx.ICON_ERROR)
+            dlg = wx.MessageBox(self, "The magnet link is invalid: %s" % str(e),
+                                "The magnet link is invalid", wx.OK | wx.ICON_ERROR)
             dlg.ShowModal()
             dlg.Destroy()
         return True

